drivers/video/ov5640.c

```diff
@@ -644,19 +890,33 @@ static int ov5640_init(const struct device *dev)
 	k_sleep(K_MSEC(5));
 
 	/* Initialize register values */
-	ret = ov5640_write_multi_regs(&cfg->i2c, ov5640InitParams, ARRAY_SIZE(ov5640InitParams));
+	ret = ov5640_write_multi_regs(&cfg->i2c, ov5640_init_params_common,
+				ARRAY_SIZE(ov5640_init_params_common));
 	if (ret) {
-		LOG_ERR("Unable to initialize the sensor");
+		LOG_ERR("Unable to initialize the sensor with common parameters");
 		return -EIO;
 	}
 
-	/* Set virtual channel */
-	ret = ov5640_modify_reg(&cfg->i2c, 0x4814, 3U << 6, (uint8_t)(DEFAULT_MIPI_CHANNEL) << 6);
+	if (ov5640_is_dvp(dev)) {
+		ret = ov5640_write_multi_regs(&cfg->i2c, ov5640_init_params_dvp,
+				ARRAY_SIZE(ov5640_init_params_dvp));
+	}
+
 	if (ret) {
-		LOG_ERR("Unable to set virtual channel");
+		LOG_ERR("Unable to initialize the sensor with DVP parameters");
 		return -EIO;
 	}
```

Comment:
This looks all correct, but it would be less confusing to have it inside the if to group it semantically.

Reply:
Done.